gdraw/ggdkdraw.c

```diff
@@ -1566,20 +1572,23 @@ static void GGDKDrawSetTransientFor(GWindow transient, GWindow owner) {
         ow = (GGDKWindow)owner;
     }
 
-    if (gw->transient_owner != NULL) {
-        g_ptr_array_remove_fast(gw->transient_owner->transient_childs, gw);
-    }
-
     if (ow != NULL) {
         gdk_window_set_transient_for(gw->w, ow->w);
         gdk_window_set_modal_hint(gw->w, true);
         gw->istransient = true;
-
-        g_ptr_array_add(ow->transient_childs, gw);
+	g_ptr_array_add(gdisp->transient_stack, gw);
+	if (gw->restrict_input_to_me)
+	    _GGDKDraw_ChangeRestrictCount(gdisp, 1);
     } else {
         gdk_window_set_modal_hint(gw->w, false);
         gdk_window_set_transient_for(gw->w, gw->display->groot->w);
         gw->istransient = false;
+
+	// Be extra careful and ensure the window was on the list, in
+	// case there are spurious remove calls.
+        // TODO: optimise; search from the back; it is meant to be a stack...
+        if ( g_ptr_array_remove(gdisp->transient_stack, gw) && gw->restrict_input_to_me )
```

Comment:
I'd recommend you run the autoformatter if you can, because I still see mixed tabs/spaces here. (although I know it indents some macros which I'd prefer it didn't, not sure how to get around that...)

Reply:
It's one thing to have different styles in different files. It's another to divide the project up into domains with independent tool chains. The project can't agree on a new style, so everything that gets added has a style determined by whoever added it?

Reply:
I still think we should format the whole code base (there was a long discussion on this somewhere), and adhere to it. The biggest drawback that was raised was losing meaningful git blame, although there are some workarounds there.

The common theme I generally see for new code is:
* Use spaces for indentation, don't mix tabs/spaces, don't dedent returns
* Egyptian braces

But yes, generally whoever wrote it will have a different flavour on that. Bugfix code usually (but not always) also follows the indentation style of the surrounding code.

As for this case in particular, all the GDK code conforms to the astyle formatter code I mentioned (barring the indentation of the ignore deprecated warning macros). You don't have to use the formatter if you don't want to, but it should generally conform to what it does.

Reply:
This should be done now.